Paid Newsletters: Implement New Design for Panels

romarioraffington opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to projects/plugins/jetpack/to-test.md in a new commit as part of your PR. -->
<!-- a12s: If you have an expected version that you're aiming for the PR to add, please use the Milestone field to communicate it. If you leave it blank, that indicates there isn't a preference. -->

Fixes https://github.com/Automattic/jetpack/issues/31216
Fixes https://github.com/Automattic/jetpack/issues/31217
Fixes https://github.com/Automattic/jetpack/issues/31218
Fixes https://github.com/Automattic/jetpack/issues/31227
Fixes https://github.com/Automattic/jetpack/issues/31219

## Proposed changes:
<!--- Explain what functional changes your PR includes -->
Implementing updated designs for Paid Newsletter. This PR will close the issues above. All issues utilize the same components hence this being one diff.

### For https://github.com/Automattic/jetpack/issues/31216

#### Designs

<img width="1412" alt="Screenshot 2023-06-06 at 1 06 18 PM" src="https://github.com/Automattic/jetpack/assets/7076981/e0746e8d-f64c-446b-a647-655a7359387d">

#### Implementation

https://github.com/Automattic/jetpack/assets/7076981/194667d2-aa4a-4fdb-821a-16a83c2dc875


### For https://github.com/Automattic/jetpack/issues/31217

#### Designs

<img width="613" alt="Screenshot 2023-06-06 at 1 06 27 PM" src="https://github.com/Automattic/jetpack/assets/7076981/358c8484-e9cc-4382-afd0-625de4b6798c">

#### Implementation

https://github.com/Automattic/jetpack/assets/7076981/0aea7aad-9ad2-4e75-971f-085be7346a04


https://github.com/Automattic/jetpack/assets/7076981/6bf8b456-a376-4eeb-8c40-0e4a152c6dfa


### For https://github.com/Automattic/jetpack/issues/31218

#### Designs

<img width="870" alt="Screenshot 2023-06-06 at 1 06 34 PM" src="https://github.com/Automattic/jetpack/assets/7076981/1f3a7e01-7ff0-4647-b58a-7d3646ef0736">

#### Implementation

https://github.com/Automattic/jetpack/assets/7076981/f63b06fe-84a8-4711-a8d9-7c7ba986246c


https://github.com/Automattic/jetpack/assets/7076981/25f3ce3d-19e5-413e-a973-e9578e4ce4d4

### For https://github.com/Automattic/jetpack/issues/31227

#### Designs

<img width="1055" alt="Screenshot 2023-06-06 at 3 49 33 PM" src="https://github.com/Automattic/jetpack/assets/7076981/759e13c5-1afd-40c6-9677-b832b66970be">

#### Implementation

https://github.com/Automattic/jetpack/assets/7076981/d18ff7af-7497-4fb8-90b2-989e2abfd883

### For https://github.com/Automattic/jetpack/issues/31219

#### Designs

<img width="392" alt="Screenshot 2023-06-09 at 3 47 12 PM" src="https://github.com/Automattic/jetpack/assets/7076981/185c5534-a859-45b4-9713-7ab762608ee7">

#### Implementation

https://github.com/Automattic/jetpack/assets/7076981/2e4f6926-fdd4-4d57-957f-7c63e469b507


### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->
No

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Compare the implementation to the designs
* Do a sanity check for the Paid Newsletter feature by publishing a newsletter post

romarioraffington committed:
Change panel name to newsletter visibility

romarioraffington committed:
Render access choices in panel instead of dropdown

romarioraffington committed:
Remove reach numbers from visibility choices

romarioraffington committed:
Remove visibility info

romarioraffington committed:
Display the newsletter notice below radio button

romarioraffington committed:
Update newsletter info and label text

romarioraffington committed:
Make notice fill entire row

romarioraffington committed:
Adjust margin and font size for access choices

romarioraffington committed:
Do not display reach text for the choice "everyone"

romarioraffington committed:
Add "also" to reach text

romarioraffington committed:
Update styles to match design

romarioraffington committed:
Update newsletter panel heading

romarioraffington committed:
Add panelHeading param to display in PrePublish panel

romarioraffington committed:
Make notice take up full width

romarioraffington committed:
Update PostPublish panel heading

romarioraffington committed:
Display the everyone notice only in the PostPublishPanel

romarioraffington committed:
changelog

romarioraffington committed:
Update setup nudge text

romarioraffington committed:
Update setup nudge styles based on design

romarioraffington committed:
Reduce margin of Stripe nudge

romarioraffington commented:
The videos above show a large gap between the Stripe nudge and the "Paid subscribers only" radio option. This has been fixed in [24ab3cf](https://github.com/Automattic/jetpack/pull/31224/commits/24ab3cf0bdec675134f36cd87a3d914ea753b66c).

<img width="1638" alt="Screenshot 2023-06-06 at 6 25 46 PM" src="https://github.com/Automattic/jetpack/assets/7076981/c3e1ad08-8453-4ab0-9965-fd3ffe96dbfa">

romarioraffington committed:
Increase specificity without !important

crisbusquets commented:
Oh, wow! That was fast, @romarioraffington ! Thanks for working on this and for sharing the videos :)

crisbusquets commented:
Btw, the post-publish panel is still WIP, we're deciding on the wording. I'll let you know once is ready 🙏🏼

romarioraffington commented:
> Btw, the post-publish panel is still WIP, we're deciding on the wording. I'll let you know once is ready 🙏🏼

Thanks @crisbusquets , putting https://github.com/Automattic/jetpack/issues/31219 and https://github.com/Automattic/jetpack/issues/31227 on hold until designs are finalized.

crisbusquets commented:
> > Btw, the post-publish panel is still WIP, we're deciding on the wording. I'll let you know once is ready 🙏🏼
> 
> Thanks @crisbusquets , putting #31219 and #31227 on hold until designs are finalized.

Hi @romarioraffington !

Now it's ready ([Figma link](h8tMpJlCqNFemEerU9owHI-fi-1788_91248)):

**Post-publish panel. Nudge if they don't have paid newsletters configured:**

Heading: `Set up a paid newsletter`
Text: `Allow your subscribers to support your work. Connect a Stripe account to get started.`
CTA: `Turn on paid newsletter`. It should link to Earn

**Post-publish panel. Nudge if they have paid newsletters, but a site plan with high fees:**

Heading: `Lower transaction fees`
Text: `With your current plan, the transaction fee for payments is XX% (+ Stripe fees). Upgrade to {another plan} to lower it.`
CTA: `Upgrade your plan`. It should link to https://wordpress.com/plans/, I think

Quick question: could we have metrics on this? Like how many people click these buttons (cc @simison).

Thank you!

romarioraffington committed:
Add spacing

simison commented:
Yes, let's have tracks events on all major CTAs like these.

romarioraffington committed:
Implement PostPublish notice design

romarioraffington commented:
> Yes, let's have tracks events on all major CTAs like these.

Added a task for this. https://github.com/Automattic/jetpack/issues/31287

romarioraffington committed:
Remove negative margin

romarioraffington commented:
> > Yes, let's have tracks events on all major CTAs like these.
> 
> Added a task for this. #31287

I'll address this in this ticket as well

millerf commented on projects/plugins/jetpack/extensions/blocks/subscriptions/settings.js:
```diff
@@ -290,10 +261,9 @@ export function NewsletterAccessDocumentSettings( {
 
 	// Immediately close the dropdown dialog after setting the post meta.
-	const setPostMetaAndClose = onClose => {
+	const setPostMetaAndClose = () => {
```
Maybe we could rename this as it does not close anymore?

millerf replied:
We could also directly use setPostMeta in the onChange? (see below)

romarioraffington replied:
Fixed in [48777e7](https://github.com/Automattic/jetpack/pull/31224/commits/48777e70de1659ef0fa8ff3a6c274f166fe90258)

millerf commented on projects/plugins/jetpack/extensions/blocks/subscriptions/settings.js:
```diff
@@ -303,59 +273,24 @@ export function NewsletterAccessDocumentSettings( {
-								/>
+								<div className="editor-post-visibility">
+									<NewsletterAccessRadioButtons
+										onChange={ setPostMetaAndClose() }
```
```suggestion
										onChange={ setPostMeta }
```

romarioraffington replied:
Fixed in [48777e7](https://github.com/Automattic/jetpack/pull/31224/commits/48777e70de1659ef0fa8ff3a6c274f166fe90258)

millerf reviewed (commented):
I proposed some small changes in the code, but it looks good otherwise.
Letting @crisbusquets give the final approval for the changes UI-wise...

romarioraffington committed:
Add "set up paid newsletter" nudge

romarioraffington committed:
Add tracks to track button click

romarioraffington committed:
Add "setPostMeta" to onChange

romarioraffington committed:
Merge branch 'trunk' into add/implement-designs-for-newsletter-panel

crisbusquets commented:
> Letting @crisbusquets give the final approval for the changes UI-wise...

Hi, thanks for the ping!

I reviewed the videos and found some UI issues:

<img width="358" alt="Screenshot 2023-06-12 at 09 22 53" src="https://github.com/Automattic/jetpack/assets/25607244/d3796f83-b202-462c-88b5-f8620cc872b2">

Are both line-height equal? The first one seems to be tighter. Can we do it like the Blaze paragraph if that's the case?

<img width="329" alt="Screenshot 2023-06-12 at 09 23 40" src="https://github.com/Automattic/jetpack/assets/25607244/6e6fa56d-909b-4b51-9934-7f315b46b60a">

The spacing below `Set up a paid newsletters` (it should be _set up a paid newsletter_, not _setup paid newsletter_) and `Promote with Blaze` is bigger than `Newsletter: All subscribers`. Can we change the latter two, so they're like the first one?

The `new window` icon appears to be a different size. The correct one is the second one.

<img width="337" alt="Screenshot 2023-06-12 at 09 26 11" src="https://github.com/Automattic/jetpack/assets/25607244/6948bf7b-1705-47b6-8c57-19e2978e1ebc">

The `Restrict your post...` was removed from Figma, because it reads negative. We can either wait for a different copy or delete it by now.

Thank you!

romarioraffington committed:
Merge branch 'trunk' into add/implement-designs-for-newsletter-panel

romarioraffington committed:
Remove restrict newsletter notice

romarioraffington committed:
Update styles based on feedback

romarioraffington committed:
Only add padding to the newsletter button

romarioraffington committed:
Increase external icon font size

romarioraffington commented:
Thanks for the @crisbusquets,

I've fixed all the following issues:

- [x] The Restrict your post... was removed from Figma, because it reads negative. We can either wait for a different copy or delete it by now.
- [x] The new window icon appears to be a different size. The correct one is the second one.
- [x] Are both line-height equal? The first one seems to be tighter. Can we do it like the Blaze paragraph if that's the case?
- [x] The spacing below Set up a paid newsletters (it should be set up a paid newsletter, not setup paid newsletter
- [ ] Promote with Blaze is bigger than Newsletter: All subscribers. Can we change the latter two, so they're like the first one?


Regarding the last change request, the font size appears to be the same for all the panel headings based on the CSS. Should I go ahead and increase the font-size?


Here's an updated video! Let me know if you have any further feedback:




https://github.com/Automattic/jetpack/assets/7076981/d7fb8a10-530c-4fac-87c3-99a9b698dafd


If this looks good to you, feel free to approve this PR

crisbusquets commented:
> Thanks for the @crisbusquets,
> 
> I've fixed all the following issues:
> 

Thanks for sharing, @romarioraffington ! Now it looks better 😊 🎉

If I'm not mistaken, the last change request should be about the spacing, not the font size. Here's a comparison between what we have implemented now vs. what you shared in the video. The right spacing is what's on the left.

![spacing](https://github.com/Automattic/jetpack/assets/25607244/89c6ad86-442c-4965-aae9-982b4a79c847)


The copy is missing an `a`: `Set up **a** paid newsletters`

romarioraffington committed:
Add missing "a"

romarioraffington committed:
Remove margin from top from paragraphs in Panel

romarioraffington committed:
changelog

romarioraffington committed:
Remove margin to match designs

romarioraffington commented:
Thanks for clarifying @crisbusquets.

> The copy is missing an a: Set up **a** paid newsletters

My bad, fixed!

> If I'm not mistaken, the last change request should be about the spacing, not the font size. Here's a comparison between what we have implemented now vs. what you shared in the video. The right spacing is what's on the left.

Should be good now 😄 


<img width="277" alt="Screenshot 2023-06-14 at 12 21 48 PM" src="https://github.com/Automattic/jetpack/assets/7076981/7e791965-bf75-4f82-9485-c8f5b494700f">

n3f commented:
Putting this here as a review helper...

- [x] Fixes https://github.com/Automattic/jetpack/issues/31216
- [x] Fixes https://github.com/Automattic/jetpack/issues/31217
- [x] Fixes https://github.com/Automattic/jetpack/issues/31218
- [x] Fixes https://github.com/Automattic/jetpack/issues/31227
- [x] Fixes https://github.com/Automattic/jetpack/issues/31219

n3f commented:
For #31216 shows no tool tip for the "Everyone" option, but I get one about the number of subscribers...

![image](https://github.com/Automattic/jetpack/assets/6354169/e0662b00-9cf8-4e8d-9566-86b817c8879c)

n3f commented:
For #31218 I noticed that the figma doesn't match when the paid plans option is striked...

![image](https://github.com/Automattic/jetpack/assets/6354169/ee476bea-7d2a-410f-834c-3672ce4ba916)

n3f commented:
Figma for #31227 to paid subscribers shows the text "sent to _n_ paid subscribers", but this is what I'm seeing:

![image](https://github.com/Automattic/jetpack/assets/6354169/a592c260-4213-413b-a1e9-1e030f251d52)

n3f commented:
Not sure if the description for #31219 matches the title. I agree that the PR fixes the title task, but the description also says that it should "Add analytics to track the button click of the CTA", but I don't see that (also not sure where to go to test...). It's likely this is OBE but wanted to highlight it.

n3f commented on projects/plugins/jetpack/extensions/blocks/subscriptions/panel.js:
```diff
@@ -184,35 +189,108 @@ function NewsletterPostPublishSettingsPanel( {
+		};
+	} );
+
+	// Tracks event when clicking on the "Turn on paid newsletter" link
+	const trackClick = useCallback(
+		() => tracks.recordEvent( 'jetpack_editor_turn_on_paid_newsletter_publish_click' ),
+		[ tracks ]
+	);
```
Oh tracks are used.  Can you provide some instructions on how to verify this worked?

romarioraffington replied:
I added the functionality, but I still have some work to do. I'm happy to remove this code if you think it would be best to have it in a separate PR.

See https://github.com/Automattic/tracks-events-registration/pull/1677  and  https://github.com/Automattic/jetpack/issues/31287 for context

n3f replied:
Yeah, I would just remove it until it's all ready to be tested.

romarioraffington replied:
Makes sense. Fixed in [4393366](https://github.com/Automattic/jetpack/pull/31224/commits/439336689a55cd66baf14babd3c4f428a9c0ec43)

n3f reviewed (changes requested):
* Does the blaze stuff belong here?  Not sure that it fits with any of the linked issues...

romarioraffington committed:
Do not render notice for everyone option

romarioraffington commented:
> For #31216 shows no tool tip for the "Everyone" option, but I get one about the number of subscribers...
> 
> ![image](https://user-images.githubusercontent.com/6354169/245903324-e0662b00-9cf8-4e8d-9566-86b817c8879c.png)

Fixed in [21552c1](https://github.com/Automattic/jetpack/pull/31224/commits/21552c132fb40e1f8debecac896c9b0a2a1af309)

romarioraffington commented:
> For #31218 I noticed that the figma doesn't match when the paid plans option is striked...
> 
> ![image](https://user-images.githubusercontent.com/6354169/245906058-ee476bea-7d2a-410f-834c-3672ce4ba916.png)

Great point. Fixed in [0a050a5](https://github.com/Automattic/jetpack/pull/31224/commits/0a050a5e1ca9e120083d2ab5b07beedd45d3615f)

romarioraffington committed:
Add subscriber type to reach notice

romarioraffington commented:
> Figma for #31227 to paid subscribers shows the text "sent to _n_ paid subscribers", but this is what I'm seeing:
> 
> ![image](https://user-images.githubusercontent.com/6354169/245907260-a592c260-4213-413b-a1e9-1e030f251d52.png)

Thanks! Fixed in [2919c1a](https://github.com/Automattic/jetpack/pull/31224/commits/2919c1a49d2ccea534b0156521507df1ab99f553)

romarioraffington commented:
> Not sure if the description for #31219 matches the title. I agree that the PR fixes the title task, but the description also says that it should "Add analytics to track the button click of the CTA", but I don't see that (also not sure where to go to test...). It's likely this is OBE but wanted to highlight it.

Thanks for pointing it out. This ticket is tracked in this https://github.com/Automattic/jetpack/issues/31287. Removed the tracks code to handle in a new PR

romarioraffington commented:
>Does the blaze stuff belong here? Not sure that it fits with any of the linked issues...

This was a request from Cris. See p1686760743804609/1686760277.716199-slack-C01B6KEJ5GE. Reverted blaze changes and created a new task. See https://github.com/Automattic/jetpack/issues/31391

romarioraffington committed:
Merge branch 'trunk' into add/implement-designs-for-newsletter-panel

romarioraffington committed:
Remove tracks

romarioraffington committed:
[not verified] Revert "Remove margin from top from paragraphs in Panel"

This reverts commit 72eb462ac4211ccf8d5df343e38b74f05951d473.

romarioraffington committed:
[not verified] Revert "changelog"

This reverts commit 180991c4182c5078ce2891297e0c4756c278d451.

romarioraffington commented:
Reverted blaze changes and created a new task. See https://github.com/Automattic/jetpack/issues/31391

romarioraffington committed:
Only show nudge in editor panel

romarioraffington committed:
Add subscriber type to NewsletterNotice

n3f reviewed (approved):
WFM 👍

simison commented:
Kudos for the work here everyone, this looks sooo much better now! 👏
